src/swap/components/SwapButton.tsx

```diff
@@ -0,0 +1,34 @@
+import { useCallback, useContext } from 'react';
+import { cn } from '../../lib/utils';
+import { SwapContext } from '../context';
+import type { SwapButtonReact } from '../types';
+
+export function SwapButton({ onSubmit }: SwapButtonReact) {
+  const {
+    onSubmit: onSubmitSwap,
+    fromAmount,
+    fromToken,
+    toAmount,
+    toToken,
+  } = useContext(SwapContext);
+
+  const handleSubmit = useCallback(() => {
+    onSubmitSwap();
+    onSubmit?.();
```

Comment:
We need to pass in something here to `onSubmit` prop. Arguably we don't do anything here and rely on `onSuccess` and `onError` instead of using this `onSubmit` prop

Reply:
yeah that was to add flexibility for the user but maybe for MVP i will just remove.